domain/ethereum.go

```diff
@@ -241,6 +363,66 @@ func (t Trace) ToProto() *protocol.TransactionEvent_Trace {
 	}
 }
 
+func TracesFromBlockData(bd *protocol.BlockData) ([]Trace, error) {
+	traces := make([]Trace, len(bd.Traces))
+	blockNumber, err := strconv.ParseInt(strings.Replace(bd.Block.Number, "0x", "", -1), 16, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	intBlockNumber := int(blockNumber)
+	for i, trace := range bd.Traces {
+		traces[i] = Trace{
+			BlockNumber: &intBlockNumber,
+			Subtraces:   int(trace.Subtraces),
+			TraceAddress: func() []int {
```

Comment:
Is this safe? I have seen before that when references/values are not provided as args to the anonymous function, the anonymous function can stick to the first value. Also I think using a function here is a bit confusing, we can do at the beginning of the loop.